presto-main/src/main/java/com/facebook/presto/operator/scalar/UrlFunctions.java

```diff
@@ -147,6 +158,37 @@ public static Slice urlExtractParameter(@SqlType("varchar(x)") Slice url, @SqlTy
         return null;
     }
 
+    @SqlNullable
+    @Description("extract raw query parameter from url")
+    @ScalarFunction
+    @LiteralParameters({"x", "y"})
+    @SqlType("varchar(x)")
+    public static Slice urlExtractRawParameter(@SqlType("varchar(x)") Slice url, @SqlType("varchar(y)") Slice parameterName)
+    {
+        URI uri = parseUrl(url);
+        if ((uri == null) || (uri.getRawQuery() == null)) {
+            return null;
+        }
+
+        Slice query = slice(uri.getRawQuery());
```

Comment:
This code seems to be duplicated from above. I think we can refactor it into a method:
```java
return extractParameter(uri.getRawQuery(), parameterName);